bar/90_contents/SyncEngine/sync.js

```diff
@@ -280,7 +280,186 @@ function(request){
               body : ['{"syncCompleted" : false}']
           };
         }
+      } else if (accessTokenSet.srcType == "Google"){
+
+        // initialization
+        var accessToken = null;
+        var host = null;
+        var port = null;
+        var user = null;
+        var pass = null;
+        var calendarId = null;
+        var refreshToken = null;
+        var pageToken = "";
+        var syncToken = "";
+        // get setting data
+        for(var i = 0; i < accessInfo.length; i++){
+          if (accessInfo[i].srcType == "Google") {
+            host = accessInfo[i].host;
+            port = accessInfo[i].port;
+            user = accessInfo[i].user;
+            pass = accessInfo[i].pass;
+            accessToken = accessInfo[i].accesstoken;
+            refreshToken = accessInfo[i].refreshtoken;
+            calendarId = accessInfo[i].calendarid;
+          }
+        }
+        if (accessTokenSet.maxSyncResults == null) {
+          maxSyncResults = initMaxSyncResults;
+        } else {
+          maxSyncResults = accessTokenSet.maxSyncResults;
+        }
+
+        try {
+          var url = "https://www.googleapis.com/calendar/v3/calendars/" + calendarId + "/events" + "?maxResults=" + maxSyncResults + "&singleEvents=true";
+          var httpClient = new _p.extension.HttpClient();
+          httpClient.setProxy(host, Number(port), user, pass);
+          var headers = {'Authorization': 'Bearer ' + accessToken};
+          var response = { status: "", headers : {}, body :"" };
+          if(null != accessTokenSet.pagetoken){
```

Comment:
Usually the condition should be the following so that undefined is also taking care of.  

    if (accessTokenSet.pagetoken)